integrations/github/jwt.go

```diff
@@ -100,21 +100,24 @@ func (i integration) getConnection(ctx context.Context) (map[string]string, erro
 func (i integration) NewClientWithAppJWTFromGitHubID(appID int64) (*github.Client, error) {
 	// Shared transport to reuse TCP connections.
 	tr := http.DefaultTransport
+	entURL, err := enterpriseURL()
+	if err != nil {
+		return nil, err
+	}
 
 	// Wrap the shared transport.
 	atr, err := ghinstallation.NewAppsTransport(tr, appID, getPrivateKey())
 	if err != nil {
 		return nil, err
 	}
+	if entURL != "" {
+		atr.BaseURL = entURL + "/api/v3"
+	}
 
 	// Initialize a client with the generated JWT injected into outbound requests.
 	client := github.NewClient(&http.Client{Transport: atr})
-	u, err := enterpriseURL()
-	if err != nil {
-		return nil, err
-	}
-	if u != "" {
-		client, err = client.WithEnterpriseURLs(u, u)
+	if entURL != "" {
```

Comment:
maybe in this case it would be more readable if we split this method for 2? enterprice and not enterprise?

Reply:
Either way I don't think we can improve much, so let's think about refactoring this later.